### PR TITLE
[rstmgr] Advance to D2

### PR DESCRIPTION
**Pull request by tjaychen**

### CODE REVIEW

File: hw/ip/rstmgr/doc/checklist.md [hunk 1]
```diff
@@ -35,25 +35,25 @@ Code Quality  | [LINT_SETUP][]                 | Done        |
...
+Documentation | [MISSING_FUNC][]        | Done        |
+Documentation | [FEATURE_FROZEN][]      | Done        |
+RTL           | [FEATURE_COMPLETE][]    | Done        |
+RTL           | [AREA_CHECK][]          | Done        |
+RTL           | [PORT_FROZEN][]         | Done        |
+RTL           | [ARCHITECTURE_FROZEN][] | Done        |
+RTL           | [REVIEW_TODO][]         | Done        |
+RTL           | [STYLE_X][]             | Done        |
+Code Quality  | [LINT_PASS][]           | Done        |
+Code Quality  | [CDC_SETUP][]           | N/A         |
```
Comment (msfschaffner) [inline hw/ip/rstmgr/doc/checklist.md:50]: Does this item have to be revisited later as well once a CDC/RDC flow is available?
If so, maybe it makes sense to mention that (e.g., like in sram_ctrl https://github.com/lowRISC/opentitan/blob/master/hw/ip/sram_ctrl/doc/checklist.md)...

**Review by msfschaffner — APPROVED** [review]

**Comment by imphil** [comment]
Can we split up this PR to do the cleanup first and see a clean nightly lint/synthesis run first before we sign off as D2?

**Comment by tjaychen** [comment]
> Can we split up this PR to do the cleanup first and see a clean nightly lint/synthesis run first before we sign off as D2?
> 
> Edit: ah sorry, I saw there is #5893 as well. So let's merge #5893 first.

yes will do!

File: hw/ip/rstmgr/doc/checklist.md (hunk 1 above)
Comment (tjaychen) [reply hw/ip/rstmgr/doc/checklist.md:50]: sounds good, let me do that.

**Pull request merged**